src/AutoUI/index.tsx

```diff
@@ -385,15 +384,20 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 	return (
 		<Box display="flex" flex={1} flexDirection="column" {...boxProps}>
 			<Spinner
-				flex={1}
 				label={
 					isBusyMessage ??
 					t('loading.resource', {
 						resource: t(`resource.${model.resource}_other`).toLowerCase(),
 					})
 				}
-				show={data == null || loading || !!isBusyMessage}
-				{...(data == null ? { width: '100%', height: '100%' } : undefined)}
+				show={
+					(Array.isArray(data)
+						? data?.length
+							? false
+							: loading
+						: data == null || loading) || !!isBusyMessage
+				}
```

Comment:
I don't really get why we didn't have an issue with this before. `loading` is repeatedly getting flipped between true and false in server-side tables. This makes it not an issue, but I'm not certain it's the right way to solve this (feels a bit like a band-aid)